sweepbatcher/presigned.go

```diff
@@ -0,0 +1,457 @@
+package sweepbatcher
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+// ensurePresigned checks that we can sign a 1:1 transaction sweeping the input.
+func (b *batch) ensurePresigned(ctx context.Context, newSweeps []*sweep) error {
+	if b.cfg.presignedHelper == nil {
+		return fmt.Errorf("presignedHelper is not installed")
+	}
+	if len(b.sweeps) != 0 {
+		return fmt.Errorf("ensurePresigned is done when adding to an " +
+			"empty batch")
+	}
+
+	sweeps := make([]sweep, len(newSweeps))
+	for i, s := range newSweeps {
+		sweeps[i] = sweep{
+			outpoint:  s.outpoint,
+			value:     s.value,
+			presigned: s.presigned,
+		}
+	}
+
+	// Cache the destination address.
+	destAddr, err := b.getSweepsDestAddr(ctx, sweeps)
+	if err != nil {
+		return fmt.Errorf("failed to find destination address: %w", err)
+	}
+
+	// Set LockTime to 0. It is not critical.
+	const currentHeight = 0
+
+	// Check if we can sign with minimum fee rate.
+	const feeRate = chainfee.FeePerKwFloor
+
+	tx, _, _, _, err := constructUnsignedTx(
+		sweeps, destAddr, currentHeight, feeRate,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to construct unsigned tx "+
+			"for feeRate %v: %w", feeRate, err)
+	}
+
+	// Check of a presigned transaction exists.
+	var batchAmt btcutil.Amount
+	for _, sweep := range newSweeps {
+		batchAmt += sweep.value
+	}
+	const presignedOnly = true
```

Comment:
maybe we could call this flag `loadOnly` or something similar?

Reply:
Renamed to `loadOnly`.